src/dmd/mars.d

```diff
@@ -923,6 +925,79 @@ private int tryMain(size_t argc, const(char)** argv)
             cgFile.write();
         }
     }
+
+    static if (SYMBOL_TRACE)
+    {
+        // this is debug code we simply hope that we will not need more
+        // then 32 Megabyte of log-buffer;
+        char* fileBuffer = cast(char*)malloc(ushort.max * 512 * 64);
+        char* bufferPos = fileBuffer;
+
+        char[255] fileNameBuffer;
+        import core.stdc.time : ctime, time;
+        auto now = time(null);
+        auto timeString = ctime(&now);
+        // replace the ' ' by _ and '\n' or '\r' by '\0'
+        {
+            int len = 0;
+            char c = void;
+            for(;;)
+            {
+                c = timeString[len++];
+                // break on null, just to be safe;
+                if (!c)
+                    break;
+
+                if (c == ' ')
+                    timeString[len - 1] = '_';
+
+                if (c == '\r' || c == '\n')
+                {
+                    timeString[len - 1] = '\0';
+                    break;
+                }
+            }
+        }
```

Comment:
I think this might be the nicer solution:
```
    import core.stdc.time : localtime, time, strftime;
    import core.stdc.stdio : printf;
    import core.thread : getpid;

    char[40] buff;
    auto now = time(null);
    strftime (buff.ptr, 40, "%Y-%m-%dT%H:%M:%S", localtime(&now));
    printf ("symbol_trace_%s_%d.log\n", buff.ptr, getpid());
```
Sorry for using phobos functions in my previous comment about this.
Note this solutiion also add's the PID, so that symbol trace files can be run in parallel, without clobering the output file.